### PR TITLE
Add Restream Support

### PR DESCRIPTION
**Pull request by vamsii777**
This PR adds support for Restream OAuth integration to the Imperial package.

Depends on 
- #100

### CODE REVIEW

**Comment by fpseverino** [comment]
You can remove the `Tests/LinuxMain.swift` file, as it is no longer needed. Also, I think you could use the Vapor CI workflow, like the one I added to [PassKit](https://github.com/vapor-community/PassKit/blob/main/.github/workflows/test.yml).

File: Sources/ImperialCore/Routing/FederatedServiceRouter.swift [hunk 1]
```diff
@@ -77,20 +77,19 @@ extension FederatedServiceRouter {
     public var errorKey: String { "error" }
     public var callbackHeaders: HTTPHeaders { [:] }
    
-    public func configureRoutes(withAuthURL authURL: String, authenticateCallback: ((Request) throws -> (EventLoopFuture<Void>))?, on router: RoutesBuilder) throws {
-		router.get(callbackURL.pathComponents, use: callback)
-		router.get(authURL.pathComponents) { req -> EventLoopFuture<Response> in
+    public func configureRoutes(withAuthURL authURL: String, authenticateCallback: ((Request) async throws -> Void)?, on router: RoutesBuilder) async throws {
+        router.get(callbackURL.pathComponents, use: self.callback)
```
Comment (fpseverino) [inline Sources/ImperialCore/Routing/FederatedServiceRouter.swift:81]:
```suggestion
        router.get(callbackURL.pathComponents, use: { try await self.callback($0) })
```
This is to remove a Sendable warning

**Review by fpseverino — CHANGES_REQUESTED** [review]
In addition to the testing and CI fix, you could update JWTKit to v5, although it may bring breaking changes.

**Comment by vamsii777** [comment]
@fpseverino, Thanks for the feedback. Given that the current PR focuses on adding a new auth provider, it might be better to handle JWT updates in a separate PR. This will keep our changes focused and easier to review. 

I'll create a new PR for updating JWTKit to v5 and address any potential breaking changes.

File: Sources/Imperial/Services/Microsoft/MicrosoftRouter.swift [hunk 2]
```diff
@@ -5,15 +5,15 @@ public class MicrosoftRouter: FederatedServiceRouter {
...
+    public let callbackCompletion: (Request, String) async throws -> Response
     public var scope: [String] = []
     public let callbackURL: String
     public var tenantID: String { Environment.get(MicrosoftRouter.tenantIDEnvKey) ?? "common" }
     public var accessTokenURL: String { "https://login.microsoftonline.com/\(self.tenantID)/oauth2/v2.0/token" }
 
     public required init(
         callback: String,
-        completion: @escaping (Request, String) throws -> (Future<ResponseEncodable>)
+        completion: @escaping (Request, String) assync throws -> Response
```
Comment (fpseverino) [inline Sources/Imperial/Services/Microsoft/MicrosoftRouter.swift:16]:
```suggestion
        completion: @escaping (Request, String) async throws -> Response
```

**Review by fpseverino — CHANGES_REQUESTED** [review]

**Comment by 0xTim** [comment]
Just an FYI I'm going to be doing an Imperial code sprint next week to go through all the issues and PRs

**Pull request closed without merging**